test/tools/yulInterpreter/EVMInstructionInterpreter.cpp

```diff
@@ -430,25 +430,37 @@ u256 EVMInstructionInterpreter::eval(
 	return 0;
 }
 
-u256 EVMInstructionInterpreter::evalBuiltin(BuiltinFunctionForEVM const& _fun, const std::vector<u256>& _arguments)
+u256 EVMInstructionInterpreter::evalBuiltin(
+	BuiltinFunctionForEVM const& _fun,
+	vector<Expression> const& _arguments,
+	vector<u256> const& _evaluatedArguments
+)
 {
 	if (_fun.instruction)
-		return eval(*_fun.instruction, _arguments);
-	else if (_fun.name == "datasize"_yulstring)
-		return u256(keccak256(h256(_arguments.at(0)))) & 0xfff;
-	else if (_fun.name == "dataoffset"_yulstring)
-		return u256(keccak256(h256(_arguments.at(0) + 2))) & 0xfff;
+		return eval(*_fun.instruction, _evaluatedArguments);
 	else if (_fun.name == "datacopy"_yulstring)
 	{
 		// This is identical to codecopy.
-		if (accessMemory(_arguments.at(0), _arguments.at(2)))
+		if (accessMemory(_evaluatedArguments.at(0), _evaluatedArguments.at(2)))
 			copyZeroExtended(
 				m_state.memory,
 				m_state.code,
-				size_t(_arguments.at(0)),
-				size_t(_arguments.at(1) & numeric_limits<size_t>::max()),
-				size_t(_arguments.at(2))
+				size_t(_evaluatedArguments.at(0)),
+				size_t(_evaluatedArguments.at(1) & numeric_limits<size_t>::max()),
+				size_t(_evaluatedArguments.at(2))
 			);
+		return 0;
+	}
+
+	string arg = std::get<Literal>(_arguments.at(0)).value.str();
```

Comment:
Why not follow the layout from `EwasmBuiltinInterpreter`, that seems to be nicer for readability?